web/_js/draw.js

```diff
@@ -380,10 +411,363 @@ function initDraw(){
+		document.getElementById("descriptionField").value = entry.description
+		document.getElementById("websiteField").value = entry.links.website.join('\n')
+		document.getElementById("subredditField").value = entry.links.subreddit.map(sub => '/r/' + sub).join('\n')
+		redoButton.disabled = true;
+		undoButton.disabled = false;
+		entryId = params.get('id')
+
+		Object.entries(entry.path).forEach(([period, path]) => {
+			period.split(", ").forEach(period => {
+				pathWithPeriods.push([period, path])
+			})
+		})
+
+	} else {
+		pathWithPeriods.push([defaultPeriod, []])
+	}
+
+	initPeriodGroups()
+
+	zoom = 4;
+
+	zoomOrigin = [
+		innerContainer.clientWidth/2  - center[0] * zoom,// + container.offsetLeft
+		innerContainer.clientHeight/2 - center[1] * zoom// + container.offsetTop
+	];
+
+	scaleZoomOrigin = [
+	   2000/2 - center[0],// + container.offsetLeft
+	   2000/2 - center[1]// + container.offsetTop
+	];
+
+   applyView();
+
+	document.addEventListener('timeupdate', (event) => {
+		renderBackground()
+		updatePeriodGroups()
+	})
+
+	periodsAdd.addEventListener('click', () => {
+		pathWithPeriods.push([defaultPeriod, []])
+		// console.log(JSON.stringify(pathWithPeriods))
+		initPeriodGroups()
+	})
+
+}
+
+function calculateCenter(path){
+
+	var area = 0,
+		i,
+		j,
+		point1,
+		point2,
+		x = 0,
+		y = 0,
+		f;
+
+	for (i = 0, j = path.length - 1; i < path.length; j=i,i++) {
+		point1 = path[i];
+		point2 = path[j];
+		f = point1[0] * point2[1] - point2[0] * point1[1];
+		area += f;
+		x += (point1[0] + point2[0]) * f;
+		y += (point1[1] + point2[1]) * f;
+	}
+	area *= 3;
+
+	return [Math.floor(x / area)+0.5, Math.floor(y / area)+0.5];
+}
+
+function initPeriodGroups() {
+
+	periodGroupElements = []
+	periodGroups.textContent = ''
+
+	// console.log(pathWithPeriods)
+
+	pathWithPeriods.forEach(([period, path], index) => {
+		let periodGroupEl = periodGroupTemplate.cloneNode(true)
+		periodGroupEl.id = "periodGroup" + index
+
+		let startPeriodEl = periodGroupEl.querySelector('.period-start')
+		let endPeriodEl = periodGroupEl.querySelector('.period-end')
+		let periodVisibilityEl = periodGroupEl.querySelector('.period-visible')
+		let periodDeleteEl = periodGroupEl.querySelector('.period-delete')
+		let periodDuplicateEl = periodGroupEl.querySelector('.period-duplicate')
+		let periodVariationEl = periodGroupEl.querySelector('.period-variation')
+
+		let [start, end, variation] = parsePeriod(period)
+		// console.log(period, start, end, variation)
+
+		startPeriodEl.id = "periodStart" + index
+		startPeriodEl.previousSibling.for = startPeriodEl.id
+		endPeriodEl.id = "periodEnd" + index
+		endPeriodEl.previousSibling.for = endPeriodEl.id
+		periodVisibilityEl.id = "periodVisibility" + index
+		periodVariationEl.id = "periodVariation" + index
+
+		startPeriodEl.value = start
+		endPeriodEl.value = end
+		startPeriodEl.max = variationsConfig[variation].versions.length - 1
+		endPeriodEl.max = variationsConfig[variation].versions.length - 1
```

Comment:
This results in the max range being limited to 100 (default max for slider).
```suggestion
		startPeriodEl.max = variationsConfig[variation].versions.length - 1
		endPeriodEl.max = variationsConfig[variation].versions.length - 1
		startPeriodEl.value = start
		endPeriodEl.value = end
```